### ISSUE TITLE
Respect vscode recommandations for Go To Symbol feature

### ISSUE THREAD

**Issue by jeandat** [microsoft/vscode#193578]
# 🚀 feature request

### Description
When using "Go to symbol" feature, search is case sensitive and do not handle fuzzy search making the whole feature almost useless. If you have to know in advance the exact name including the case, it is way more useless.

### Describe the solution you'd like
As mentionned by the team in [this related ticket](https://github.com/microsoft/vscode/issues/33746), it is the plugin responsability to interpret the search in a relaxed way.

> The query-parameter should be interpreted in a relaxed way as the editor will apply its own highlighting and scoring on the results. A good rule of thumb is to match case-insensitive and to simply check that the characters of query appear in their order in a candidate symbol. Don't use prefix, substring, or similar strict matching.

I might be wrong, but it seems to me this is this plugin responsability then.

**Comment by jrieken** [comment]
This issue lacks

* naming the language and extensions providing the results

**Comment by jeandat** [comment]
My bad: Typescript language and extension

**Comment by jrieken** [comment]
@mjbvz Why do you give this back to me? Isn't it TS doing the extra filtering? All renderer-side filtering is relaxed

**Comment by mjbvz** [comment]
@jrieken Do we have any guidelines for how languages are supposed to implement this? It's one of the few features where the languages implementations do the filtering instead of VS Code

**Comment by jrieken** [comment]
Yes, it's explained here: "_[A good rule of thumb is to match case-insensitive and to simply check that the](https://github.com/microsoft/vscode/blob/623feb6ddd360d97dc77e2ff577dd2c549ef442d/src/vscode-dts/vscode.d.ts#L3596-L3598)_"
